feyton/atlp-backend
# [#180984438] - Improve documentation and add additional tests

#### What does this PR do?
This PR adds:
- The testing suite for the blog routes
- Rich documentation with swagger
- Refactored and highly maintainable code
- Requests and Error Logging

This PR Fixes:
- Code repetition
- Missing error handling
- 

#### Description of Task to be completed?
This PR is the major upgrade ready to be pushed into a testing environment.
With rich documentation, this PR makes project structure more understandable and 
usable by people. 
You can check the PR on Heroku on our testing branch [Visit Heroku here](https://atlp-backend-staging.herokuapp.com/docs/#/Blog/get_api_v1_blogs)
#### How should this be manually tested?
Nothing changed in the underlying testing requirements.
Refer to our README and remember to create a .env file to keep your environment variables. Make sure that at least it has these configured:
**.ENV**
```
ACCESS_TOKEN_SECRET=
REFRESH_TOKEN_SECRET=
MONGO_DB_URL = 
TESTING_DB_URL=
API_BASE= /api/v1
NODE_ENV="dev"
JWT_EXPIRATION=60
JWT_REFRESH_EXPIRATION=1200
SERVER_URL=http://127.0.0.1:3500
SERVER_NAME="LOCAL HOST"
```
#### What are the relevant pivotal tracker stories?
These are the relevant stories:
- [#180969915](https://www.pivotaltracker.com/story/show/180969915)
- [#180971362](https://www.pivotaltracker.com/story/show/180971362)
